File: backend/FragranceCommerce.Api/Models/Review.cs
namespace FragranceCommerce.Api.Models;

public class Review : BaseEntity
{
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    public Guid ProductId { get; set; }
    public Product Product { get; set; } = null!;
    public int Rating { get; set; }
    public string? Title { get; set; }
    public string? Comment { get; set; }
    public ICollection<ReviewImage> Images { get; set; } = new List<ReviewImage>();
}

File: frontend/fragrance-commerce-web/src/types/review.ts
export interface ReviewImage {
    id: string;
    imageUrl: string;
    displayOrder: number;
}

export interface Review {
    id: string;
    userId: string;
    userName: string;
    rating: number;
    title?: string | null;
    comment?: string | null;
    createdAt: string;
    isVerifiedPurchase: boolean;
    images: ReviewImage[];
}

export interface CreateReviewRequest {
    rating: number;
    title?: string;
    comment?: string;
    images: {
        imageUrl: string;
        displayOrder: number;
    }[];
}